### DISCUSSION TITLE
Thumbnail image의 alt data는 어디에 저장하고 어디서 처리해야 좋을까요?

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-teams/2022-moamoa#17] [토론]
지금은 간단히 프론트에서 타이틀 데이터에 "스터디 이미지" 스트링만 붙여서 alt로 사용하는데, 이거를 DB에 저장해서 그냥 바로 보내주는게 맞는지?  
아니면 지금처럼 그냥 프론트에서 처리해주는게 좋을지...?

**Comment by tco0427** [comment]
현재 DB에 스터디에 대한 `타이틀`이 저장되어 있는 상황이고, `스터디 이미지` 라는 단순 문자열을 연결하는 처리(concatenation)라면 Front에서 처리해주는게 좋을 것 같다.
굳이 DB의 컬럼을 하나 늘리는 것도 무의미하고, NW를 타서 데이터를 전달해줄 필요성을 느끼지 못했다.
(각각의 클라이언트에서 문자열을 추가해주는 처리를 해주면 되는데, 중앙집중식이라는 가정하에 서버에서 한 클라이언트 한 클라이언트 모두에 대한 처리를 해주다보면 부하가 커질 수 있다고 생각한다.)

그런데, 만약 사용자가 `직접` 이미지에 대한 설명을 추가해주는 경우라면 이 또한 `의미있는` 데이터가 되기 때문에 (`타이틀`이라는 정보와는 별개로) 별도로 DB에 저장해주는 것이 좋을 것 같다.

즉, 질문 준 상황과 같은 상황이라면 Front에서 처리해주어도 좋을 것 같다!

**Comment by sc0116** [comment]
만약 사용자가 alt 설명에 대해서 적을 수 있으면 충분히 관리할 가치가 있는 데이터라고 생각해서 DB에 저장하는게 좋다고 생각하지만, 단순히 타이틀 + "썸네일 이미지입니다." 정도면 굳이 저장할 필요가 있을까 싶다!

**Comment by jaejae-yoo** [comment]
- 사용자의 데이터를 받는 등 **데이터가 가변적으로 변하는 경우 DB에 저장**해야 한다고 생각합니다.

- **제목**에 "**이미지**" 텍스트만 붙여서 보여주는 경우, DB에 저장해야 할 이점이 명확하지 않다고 생각합니다.

**Comment by nan-noo** [comment]
항상 alt양식이 고정되어 있으면 프론트에서 조합하든 백엔드에서 조합해서 응답으로 보내주든 큰 차이가 없을 것 같아요! 양식이 고정되어 있으면 있는 데이터 조합해서 보내주기만 하면 되니까 DB에 저장할 필요도 없을 것 같네용

대형 쇼핑몰에선 이미지 알트 양식이 정해지지 않았을 것 같은데, 그런 경우엔 사용자 입력값 사용, AI로 분석, 수동으로 입력(?) 등을 하지 않을까요?? 이럴때는 DB에 저장하는 게 맞을 것 같아요!

**Comment by airman5573** [comment]
상황에 따라 다를것 같습니다.

1. alt를 사용자가 입력하는 경우 : 가변적이기 때문에 DB에 저장하는게 맞다고 생각합니다.  
2. alt가 고정적인 경우(혹은 기존 DB에 있는 데이터로 조합 할 수 있는 경우)에는 DB에 없어도 된다고 생각합니다.
